pkg/controller/replication_controller.go

```diff
@@ -327,6 +327,7 @@ func (rm *ReplicationManager) syncReplicationController(key string) error {
 	obj, exists, err := rm.controllerStore.Store.GetByKey(key)
 	if !exists {
 		glog.Infof("Replication Controller has been deleted %v", key)
+		rm.expectations.DeleteExpectations(key)
```

Comment:
OK I thought about it and I think that since we require the client to wait anyway, it's OK if RC manager doesn't do this for the moment.